jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+{:.no_toc}
+
+An orb `foo` might define a job:
+
+```yaml
+# yaml from orb `foo`
+jobs:
+  bar:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          command: echo "building"
+      - run:
+          command: echo "testing"
+```
+
+Then an orb user could use the job as follows:
+```yaml
+# config.yml
+version: 2.1
+orbs:
+  foo: somenamespace/foo@1.2
+workflows:
+  build:
+    jobs:
+      - foo/bar:
+          pre-steps:
+            - run:
+                command: echo "install custom dependency"
+          post-steps:
+            - run:
+                command: echo "upload artifact to s3"
+```
+
+The resulting configuration would look like this:
+
+```yaml
+version: 2.1
+jobs:
+  foo/bar:
+    machine: true
+    steps:
+      - run:
+          command: echo "install custom dependency"
+      - checkout
+      - run:
+          command: echo "building"
+      - run:
+          command: echo "testing"
+      - run:
+          command: echo "upload artifact to s3"
+workflows:
+  build:
+    jobs:
+      - foo/bar
+```
+
+## Conditional Steps
+**Note:** Conditional steps are available in configuration version 2.1 and later.
+
+Conditional steps allow the definition of steps that only run if a condition is
+met. 
+
+<!---
+For example, an orb could define a command that runs a set of steps *if* the
+orb's user invokes it with `myorb/foo: { dostuff: true }`, but not
+`myorb/foo: { dostuff: false }`.
+-->
+
+These conditions are checked before a workflow is actually run. That
+means, for example, that a user can't use a condition to check an environment
+variable.
+
+Conditional steps can be located anywhere a regular step could. 
+
+<!---
+For example, an
+orb author could define conditional steps in the `steps` key of a Job or a
+Command.
+-->
+
+A conditional step consists of a step with the key `when` or `unless`. Under this conditional key are the subkeys `steps` and `condition`. If `condition` is met (using when/unless logic), the subkey `steps` are run. 
+
+A `condition` is a single value that evaluates to `true` or `false` at the time the config is processed, so you cannot use environment variables as conditions, as those are not injected until your steps are running in the shell of your execution environment. You may use parameters as your conditions. The empty string will resolve as falsey in `when` conditions.
+
+### Example
+{:.no_toc}
+
+```yaml
+# Contents of the orb `myorb` in namespace `mynamespace`
+jobs:
+  myjob:
+    parameters:
+      preinstall-foo:
+        type: boolean
+        default: false
+    machine: true
+    steps:
+      - run: echo "preinstall is << parameters.preinstall-foo >>"
+      - when:
+          condition: << parameters.preinstall-foo >>
+          steps:
+            - run: echo "preinstall"
+      - unless:
+          condition: << parameters.preinstall-foo >>
+          steps:
+            - run: echo "don't preinstall"
+```
+
+```yaml
+# inside config.yml
+version: 2.1
+
+orbs:
```

Comment:
see above comment about using orbs in these examples

Reply:
To make this example work without orbs:
- the above and below yaml blocks can be merged into 1
  - the comment `Contents of the orb myorb in namespace mynamespace` can be deleted
- under `workflow:`, `myorb/myjob` => `myjob`
- the `orbs` section can be deleted

Reply:
cool, thanks I merged these